library/common/types/c_types.cc

```diff
@@ -1,6 +1,15 @@
+// NOLINT(namespace-envoy)
 #include "library/common/types/c_types.h"
 
-#include <string.h>
+#include <string>
+
+#include "common/common/assert.h"
+
+void* safe_malloc(size_t size) {
+  void* ptr = malloc(size);
+  RELEASE_ASSERT(ptr != nullptr, "malloc failure");
```

Comment:
...use `exit(EXIT_FAILURE)`

Reply:
Also you need to check that `size > 0` before crashing.

Reply:
Ah yes, good call. I keep forgetting that is a possibility re: `size > 0` check.